scripts/orchestra_methods.py

```diff
@@ -0,0 +1,319 @@
+    """Map the initial inputs to CellRanger Count.
+        Arguments:
+        paths- gsurls to every fastq directory
+        masterCsv- Csv File mapping samples, lanes and indices to bcl
+        sampleIds- list of every sample, not actually needed, just easier
+        transMap- map of reference name to gsurl of transcriptome files
+        Outputs-
+        paths.tsv- Cromwell mapping of sample to a comma seperated string of fastq gsurls
+        reference_list.tsv- Cromwell mapping of sample to genome
+        transcriptome_list.tsv- Cromwell mapping of sample to transcriptome file
+        chemistry.tsv- Cromwell mapping of sample to chemistry
+    """
+    # get a list of every fastq/sample directory using gsutil ls
+    fastqs = []
+    for f in paths:
+        fastqs = fastqs + check_output(["gsutil", "ls", f]).decode("utf-8").split("\n")
+    # open the files for writing
+    chemistry_list = open('chemistry.tsv', 'w+')
+    reference_list = open('reference.tsv', 'w+')
+    transcriptome_list = open('transcriptome.tsv', 'w+')
+    paths_list = open('paths.tsv', 'w+')
+    # open the masterCsv and transcriptome map csv
+    df = pd.read_csv(masterCsv)
+    tm = pd.read_csv(transMap)
+    # create the maps
+    for sample in sampleIds:
+        # Sample paths map
+        # a path matches if it ends with .../sample_id/
+        key = os.path.join(sample,"")
+        filter_paths = ",".join([path for path in fastqs if path.endswith(key)])
+        # Write to file
+        paths_list.write("%s\t%s\n" % (sample, filter_paths))
+        # Chemistry and Genome map
+        # Chemistry and Genome are the same across sample (assumed) so we can just use the first one we get
+        rows = df.loc[df["Sample"] == sample]
+        chemistry = rows["Chemistry"].tolist()[0]
+        # first get the genome for the sample
+        reference = rows["Reference"].tolist()[0]
+        # we have a map of reference names to gsurls
+        transcriptome_file = list(tm[tm["Reference Name"] == reference]["Location"])[0]           
+        # Write to files
+        chemistry_list.write("%s\t%s\n" % (sample, chemistry))
+        reference_list.write("%s\t%s\n" % (sample, reference))
+        transcriptome_list.write("%s\t%s\n" % (sample, transcriptome_file))
+    # close the files
+    transcriptome_list.close()
+    reference_list.close()
+    chemistry_list.close()
+    paths_list.close()
+
+def __main__(argv):
+    """Command Line parser for scRna-Seq pipeline.
+    Inputs- command line arguments
+    """
+    # the first command, -c, tells us which arguments we need to check for next, and which function we are going to call
+    command = argv[1].replace("-c=", "")
+    # create the argument parser
+    parser = argparse.ArgumentParser()
+    if command == "count":
+        # CellRanger count method
+        # add arguments
+        parser.add_argument('--sampleId', '-id', help="Id of sample being run", type= str)
```

Comment:
`type= str` is a default for `add_argument`.  I recommend omitting it here and elsewhere, and only using `type` when the value is not a default (i.e., not a string).

Reply:
Fixed